### PR TITLE
build(moment): Upgrade moment

### PR DESCRIPTION
**Pull request by leedongwei**
Upgrade for fix in `v2.28.0`: https://github.com/moment/moment/blob/develop/CHANGELOG.md#2280-see-full-changelog

### Before (v2.24.0)
```
> moment('2021-04-01T00:00:00.000').format()
  "2021-04-01T00:00:00Z"    // parsed to UTC instead of local
> moment.utc('2021-04-01T00:00:00.000').format()
  "2021-04-01T00:00:00Z"
```

### After (v2.29.1)
```
> moment('2021-04-01T00:00:00.000').format()
  "2021-04-01T00:00:00-07:00"
> moment.utc('2021-04-01T00:00:00.000').format()
  "2021-04-01T00:00:00Z"
```

### CODE REVIEW

**Comment by leedongwei** [comment]
Looks like there's some type errors. Switching to draft.

File: static/app/components/dateTime.tsx [hunk 1]
```diff
@@ -84,8 +85,8 @@ class DateTime extends React.Component<Props> {
     return (
       <time {...carriedProps}>
         {utc
-          ? moment.utc(date).format(format)
-          : moment.tz(date, options?.timezone ?? '').format(format)}
+          ? moment.utc(date as moment.MomentInput).format(format)
```
Comment (EvanPurkhiser) [inline static/app/components/dateTime.tsx:88]: Is this a safe cast?

Comment (leedongwei) [reply static/app/components/dateTime.tsx:88]: The moment type file changed from `null | undefined` to `void` ([see commit](https://github.com/moment/moment/commit/5a1843de5a98025aa04437df8d7e4e7200306704)), which is more accurate. It'll work just fine with the casting.

**Review by evanpurkhiser — APPROVED** [review]

**Comment by billyvg** [comment]
Did you do any manual testing? I think a quick test with a local + utc timezone (via user settings) on Discover would be good before merging

**Comment by leedongwei** [comment]
@billyvg Yep, I'm doing so on `staging`.

**Pull request merged**